okhttp/src/main/java/io/grpc/okhttp/OkHttpServerBuilder.java

```diff
@@ -0,0 +1,376 @@
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ServerSocketFactory;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.x500.X500Principal;
+
+/** Build servers with the OkHttp transport. */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/1785")
+public final class OkHttpServerBuilder extends ForwardingServerBuilder<OkHttpServerBuilder> {
+  private static final Logger log = Logger.getLogger(OkHttpServerBuilder.class.getName());
+  private static final int DEFAULT_FLOW_CONTROL_WINDOW = 65535;
+  private static final long AS_LARGE_AS_INFINITE = TimeUnit.DAYS.toNanos(1000L);
+  private static final ObjectPool<Executor> DEFAULT_TRANSPORT_EXECUTOR_POOL =
+      OkHttpChannelBuilder.DEFAULT_TRANSPORT_EXECUTOR_POOL;
+
+  /**
+   * Always throws, to shadow {@code ServerBuilder.forPort()}.
+   *
+   * @deprecated Use {@link #forPort(int, ServerCredentials)} instead
+   */
+  @DoNotCall("Always throws. Use forPort(int, ServerCredentials) instead")
+  @Deprecated
+  public static OkHttpServerBuilder forPort(int port) {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Creates a builder for a server listening on {@code port}.
+   */
+  public static OkHttpServerBuilder forPort(int port, ServerCredentials creds) {
+    return forPort(new InetSocketAddress(port), creds);
+  }
+
+  /**
+   * Creates a builder for a server listening on {@code address}.
+   */
+  public static OkHttpServerBuilder forPort(SocketAddress address, ServerCredentials creds) {
+    HandshakerSocketFactoryResult result = handshakerSocketFactoryFrom(creds);
+    if (result.error != null) {
+      throw new IllegalArgumentException(result.error);
+    }
+    return new OkHttpServerBuilder(address, result.factory);
+  }
+
+  final ServerImplBuilder serverImplBuilder = new ServerImplBuilder(this::buildTransportServers);
+  final SocketAddress listenAddress;
+  final HandshakerSocketFactory handshakerSocketFactory;
+  TransportTracer.Factory transportTracerFactory = TransportTracer.getDefaultFactory();
+
+  ObjectPool<Executor> transportExecutorPool = DEFAULT_TRANSPORT_EXECUTOR_POOL;
+  ObjectPool<ScheduledExecutorService> scheduledExecutorServicePool =
+      SharedResourcePool.forResource(GrpcUtil.TIMER_SERVICE);
+
+  ServerSocketFactory socketFactory = ServerSocketFactory.getDefault();
+  long keepAliveTimeNanos = KEEPALIVE_TIME_NANOS_DISABLED;
+  long keepAliveTimeoutNanos = DEFAULT_KEEPALIVE_TIMEOUT_NANOS;
+  int flowControlWindow = DEFAULT_FLOW_CONTROL_WINDOW;
+  int maxInboundMetadataSize = GrpcUtil.DEFAULT_MAX_HEADER_LIST_SIZE;
+  int maxInboundMessageSize = GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE;
```

Comment:
These members have setters to populate them, but why don't I see them used anywhere?

Reply:
I could believe some of these are as-of-yet unused and are part of some TODO/FIXME/should-add-a-TODO. But this particular field is used in `OkHttpServerTransport.Config` and then copied to `OkHttpServerStream.TransportState`.